src/Optimizely/OptimizelyConfig/OptimizelyConfigService.php

```diff
@@ -69,6 +81,10 @@ public function __construct(ProjectConfigInterface $projectConfig)
         $this->featureFlags = $projectConfig->getFeatureFlags();
         $this->revision = $projectConfig->getRevision();
         $this->datafile = $projectConfig->toDatafile();
+        $this->environmentKey = $projectConfig->getEnvironmentKey();
+        $this->sdkKey = $projectConfig->getSdkKey();
+        $this->projectConfig = $projectConfig;
```

Comment:
Why is this required here now? can we have the methods which required project config inside it?

Reply:
in this way our getfeatureMap and getExprimentMap will require an additional argument which can disturb the existing flow so idea is to keep them as they are and using projectConfig in new functions inside